tests/dojo_tests.py

```diff
@@ -1,40 +1,132 @@
-        self.assertEqual(log,"\nCannot creae room(s), invalid room type enterred")
-    def test_check_duplicate_names(self):
-        ndojo = Dojo()
-        log = ndojo.create_room(["Green", "Blue"],"office")
-        log = ndojo.create_room(["Blue"],"office")
-        self.assertEqual(log,"\nThe name of office at index 0 already existed.")
-        log = ndojo.create_room(["Brown","Black","Black"],"livingspace")
-        self.assertEqual(log,"\nThe name of office at index 2 already existed.")
+        new_room_count = len(self.ndojo.all_rooms)
+        self.assertEqual(new_room_count - initial_room_count, 3)
+        self.ndojo.create_room(["Yellow", "Brown", "Black"], "livingspace")
+        latest_room_count = len(self.ndojo.all_rooms)
+        self.assertEqual(latest_room_count - new_room_count, 3)
+
+    def test_check_empty_input(self):
+        """ This function test for empty room name and room type inputs """
+        self.ndojo.create_room([], "  ")
+        log = sys.stdout.getvalue().strip()
+        self.assertEqual(log, \
+            "Cannot create rooms with empty room name and/or empty room type")
+
+    def test_invalid_room_input(self):
+        """ This function test for empty room name"""
+        self.ndojo.create_room([" "], "office")
+        log = sys.stdout.getvalue().strip()
+        self.assertEqual(log, \
+            "The office at index 0 cannot be created due to empty name.")
+
+    def test_invalid_in_array_input(self):
+        """ This function test for empty room name in array rooms"""
+        self.ndojo.create_room(["Green", " ", "Black"], "livingspace")
+        log = sys.stdout.getvalue().strip()
+        log = log.split("\n")
+        self.assertEqual(log[len(log) - 1], \
+           "The livingspace at index 1 cannot be created due to empty name.")
+
+    def test_invalid_room_type(self):
+        """ This function test for invalid room type """
+        self.ndojo.create_room(["Green", "Black"], "piper")
+        log = sys.stdout.getvalue().strip()
+        self.assertEqual(log, \
+            "Cannot create room(s), invalid room type enterred")
+
+    def test_duplicate_office_name(self):
+        """ This function test for creation of duplicate office names """
+        self.ndojo.create_room(["Green", "Blue"], "office")
+        self.ndojo.create_room(["Blue"], "office")
+        log = sys.stdout.getvalue().strip()
+        log = log.split("\n")
+        self.assertEqual(log[len(log) - 1], "The office at index 0 already existed.")
+
+    def test_duplicate_livingspace_name(self):
+        """ This function test for creation of duplicate livingspace names """
+        self.ndojo.create_room(\
+                    ["Brown", "Black", "Black"], "livingspace")
+        log = sys.stdout.getvalue().strip()
+        log = log.split("\n")
+        self.assertEqual(log[len(log) - 1], "The livingspace at index 2 already existed.")
+
+    def test_print_rooms(self):
```

Comment:
Do you think it would provide better structure to put these tests that check `printing of stuff` in their own test case other than in the `TestCreateRoom` testcase?